pkg/controller/controller.go

```diff
@@ -563,8 +567,16 @@ func (p *csiProvisioner) ProvisionExt(options controller.ProvisionOptions) (*v1.
 
 	ctx, cancel := context.WithTimeout(context.Background(), p.timeout)
 	defer cancel()
+	startTime := time.Now()
 	rep, err = p.csiClient.CreateVolume(ctx, &req)
 
+	p.metricsManager.RecordMetrics(
```

Comment:
Would do you think of adding the metrics in the common grpc handler, so that we automatically record metrics for every single csi call?

https://github.com/kubernetes-csi/external-provisioner/blob/master/vendor/github.com/kubernetes-csi/csi-lib-utils/connection/connection.go#L179

Reply:
That seems like a good idea.

Reply:
I'll work on another PR to do this. If I can finish that in the next couple of days, we can abandon this PR. Otherwise, we can merge this to unblock the larger effort.